apis/python/src/tiledbsoma/tiledb_platform_config.py

```diff
@@ -1,5 +1,9 @@
 from dataclasses import dataclass
-from typing import Optional
+from typing import Any, Dict, Iterator, Mapping, Optional, Sequence, Type, Union, cast
+
+import tiledb
+
+from .types import PlatformConfig
 
 
 @dataclass(frozen=True)
```

Comment:
I believe many of the values in this class are obsolete and could be deleted.  Several only make sense in `main-old`, or are obsoleted by this new work (with new defaulting).

I believe the obsolete items are:  X_capacity, X_tile_order, obs_extent, var_extent, and string_dim_zstd_level

Reply:
More things _should be brought over from_ `main-old`

Reply:
That said, we can strip down to what's actually in use in today's foundational classes in `main`, then add to `platform_config` as we add to those classes

Reply:
I agree, but they need to be reworked and moved elsewhere.   They are mostly soma.io preferences/defaults, not DataFrame, SparseNdArray, or DenseNdArray.  Ie, they don't belong in this file.

Reply:
All good! I remember they were added "for reasons" back in the `main-old` days, so I just want to make sure those reasons aren't lost -- thanks for checking and reassuring me! :)

Reply:
This seems like something that should get taken care of in a follow-up.

Reply:
That is completely OK with me.  I just personally like to stay on top of detritus :-)